stdlib/public/core/RangeReplaceableCollection.swift

```diff
@@ -872,6 +872,39 @@ extension RangeReplaceableCollection where Self : BidirectionalCollection {
     let end = endIndex
     removeSubrange(index(end, offsetBy: -k)..<end)
   }
+
+  /// Removes from the end of the collection and returns the specified
+  /// number of elements.
+  ///
+  /// Attempting to extract more elements than exist in the collection
+  /// triggers a runtime error.
+  ///
+  /// Calling this method may invalidate all saved indices of this
+  /// collection. Do not rely on a previously stored index value after
+  /// altering a collection with any operation that can change its length.
+  ///
+  /// - Parameter k: The number of elements to extract from the collection.
+  ///   `k` must be greater than or equal to zero and must not exceed the
+  ///   number of elements in the collection.
+  ///
+  /// - Complexity: O(*k*), where *k* is the specified number of elements.
+  @inlinable
+  public mutating func extractLast(_ k: Int) -> Self {
+    // Duplicate removeLast(_:) logic above to avoid redundant preconditions
+    // and index calculation
+    if k == 0 { return Self() }
+    _precondition(k >= 0, "Number of elements to remove should be non-negative")
```

Comment:
This should be `k > 0` as the check above takes care of the scenario where k == 0.